mirrord/protocol/src/codec.rs

```diff
@@ -127,6 +129,8 @@ pub enum FileResponse {
     GetDEnts64(RemoteResult<GetDEnts64Response>),
     ReadLink(RemoteResult<ReadLinkFileResponse>),
     ReadDirBatch(RemoteResult<ReadDirBatchResponse>),
+    MakeDir(RemoteResult<MakeDirResponse>),
+    MakeDirAt(RemoteResult<MakeDirResponse>),
```

Comment:
These could be merged into one variant (like `FileRequest::Open` and `FileRequest::OpenRelative` both result in `FileResponse::Open`).
Additionaly, `MakeDirResponse` contains only error code, which should already be handled by `RemoteResult` wrapper. So the variant can look like:
```rust
pub enum FileResponse {
  ...
  
  MakeDir(RemoteResult<()>),
}
```

Reply:
By using the MakeDirResponse, all error codes were returned to the mirrord client. I think that not all [mkdir](https://man7.org/linux/man-pages/man2/mkdir.2.html)/libc error codes are mapped on error.rs if we go for the ResponseError option. I think this also applies to other libc functions, Are we ok with that?

Reply:
Yeah, that's ok ^^